lightly/transforms/add_grid_transform.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright (c) Lightly AG and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+#
+from math import ceil
+from typing import Any, Dict, Tuple
+
+import torch
+from torchvision.transforms.v2 import CenterCrop, ConvertBoundingBoxFormat, Transform
+from torchvision.tv_tensors import BoundingBoxes, BoundingBoxFormat, Mask
+
+from lightly.utils import dependency as _dependency
+
+
+# ignore typing due to Any type used in torchvison.transforms.v2.Transform
+class AddGridTransform(Transform):  # type: ignore[misc]
+    """Implements the naive segmentation into a regular grid from DetCon. [0]
+
+    Input to this transform:
```

Comment:
I was mainly following the other transforms, e.g. [BYOL](https://github.com/lightly-ai/lightly/blob/master/lightly/transforms/byol_transform.py), [MAE](https://github.com/lightly-ai/lightly/blob/master/lightly/transforms/mae_transform.py). But I am more than happy to start transitioning to `Args:` and `Returns:`, maybe even do it a bit more proper and put them inside a separate docstring of the `__call__` or `forward` method.

Reply:
I see, note that these transforms are special because they combine multiple transforms into one for convenience. The transform you're implementing here however is standalone (see e.g. `rotation.py` for reference). However, you're right that we don't have a `__call__` or `forward` function to annotate so let's keep it the way it is for now.